openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -32,6 +38,111 @@ def get_identity(self, data):
             return None
 
 
+class UserProfileReadOnlySerializer(serializers.Serializer):
+    """
+    Class that serializes the portion of User model needed for account information.
+    """
+    def __init__(self, *args, **kwargs):
+        # Don't pass the 'configuration' arg up to the superclass
+        self.configuration = kwargs.pop('configuration', None)
+        if not self.configuration:
+            self.configuration = settings.ACCOUNT_VISIBILITY_CONFIGURATION
+
+        # Don't pass the 'custom_fields' arg up to the superclass
+        self.custom_fields = kwargs.pop('custom_fields', None)
+
+        super(UserProfileReadOnlySerializer, self).__init__(*args, **kwargs)
+
+    def to_native(self, user):
+        """
+        Overwrite to_native to handle custom logic since we are serializing two models as one here
+        :param user: User object
+        :return: Dict serialized account
+        """
+        profile = user.profile
+
+        data = {
+            "username": user.username,
+            "url": self.context.get('request').build_absolute_uri(
+                reverse('accounts_api', kwargs={'username': user.username})
+            ),
+            "email": user.email,
+            "date_joined": user.date_joined,
+            "is_active": user.is_active,
+            "bio": AccountLegacyProfileSerializer.convert_empty_to_None(profile.bio),
+            "country": AccountLegacyProfileSerializer.convert_empty_to_None(profile.country.code),
+            "profile_image": self._get_profile_image(profile, user),
+            "time_zone": None,
+            "language_proficiencies": LanguageProficiencySerializer(
+                profile.language_proficiencies.all(),
+                many=True
+            ).data,
+            "name": profile.name,
+            "gender": AccountLegacyProfileSerializer.convert_empty_to_None(profile.gender),
+            "goals": profile.goals,
+            "year_of_birth": profile.year_of_birth,
+            "level_of_education": AccountLegacyProfileSerializer.convert_empty_to_None(profile.level_of_education),
+            "mailing_address": profile.mailing_address,
+            "requires_parental_consent": profile.requires_parental_consent(),
+        }
+
+        return self._filter_fields(
+            self._visible_fields(profile, user),
+            data
+        )
+
+    def _get_profile_image(self, user_profile, user):
+        """ Returns metadata about a user's profile image. """
+        data = {'has_image': user_profile.has_profile_image}
```

Comment:
it does but it works slightly differently. It takes an additional `user` argument and passes the `request` to the profile image builder.  I wanted to keep any changes out of the legacy serializers since they are still used for updates so had this class implement them and made the changes there.  I think that there is a lot of more refactor work that can/should happen here.  If we can make this serializer work in both directions (read & write) then we can remove the two separate User and UserProfile serializers.  For now that seems out of scope for this fix.

```python
    def get_profile_image(self, user_profile):
        """ Returns metadata about a user's profile image. """
        data = {'has_image': user_profile.has_profile_image}
        urls = get_profile_image_urls_for_user(user_profile.user)
        data.update({
            '{image_key_prefix}_{size}'.format(image_key_prefix=PROFILE_IMAGE_KEY_PREFIX, size=size_display_name): url
            for size_display_name, url in urls.items()
        })
        return data
```